### PR TITLE
fix backwards compatibility issues in legacySystemSCs

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: epochStart/metachain/legacySystemSCs.go [hunk 1]
```diff
@@ -752,14 +754,28 @@ func (s *legacySystemSCProcessor) stakingToValidatorStatistics(
 	}
 
 	newValidatorInfo := s.validatorInfoCreator.PeerAccountToValidatorInfo(account)
-	err = validatorsInfoMap.Replace(jailedValidator, newValidatorInfo)
+	err = s.replaceValidators(jailedValidator, newValidatorInfo, validatorsInfoMap)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply epochStart/metachain/legacySystemSCs.go:757]: I have to agree here. let's do the code without returning error. and the same with unStakeNodesWithNotEnoughFunds - if we cannot replace - we keep that old validator and it should be good.

**Review by sasurobert — CHANGES_REQUESTED** [review]

File: epochStart/metachain/legacySystemSCs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply epochStart/metachain/legacySystemSCs.go:757]: Yes, as mentioned when opening the PR, I wanted to discuss the errors that have been added here by @mariusmihaic. My preference was also to not change the initial behavior (in regards to the new errors being returned). 

The code here was to have something to start the discussion and see that the backwards issue is solved.

Will remove the errors for v4 as well.

File: state/interface.go [hunk 2]
```diff
@@ -292,7 +293,9 @@ type ShardValidatorsInfoMapHandler interface {
 
 	Add(validator ValidatorInfoHandler) error
 	Delete(validator ValidatorInfoHandler) error
+	DeleteKey(blsKey []byte, shardID uint32)
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply state/interface.go:296]: done

File: epochStart/metachain/legacySystemSCs.go [hunk 3]
```diff
@@ -752,14 +754,28 @@ func (s *legacySystemSCProcessor) stakingToValidatorStatistics(
...
+	old state.ValidatorInfoHandler,
+	new state.ValidatorInfoHandler,
+	validatorsInfoMap state.ShardValidatorsInfoMapHandler,
+) error {
+	stakingV4Enabled := s.enableEpochsHandler.IsFlagEnabled(common.StakingV4StartedFlag)
+	if stakingV4Enabled {
+		return validatorsInfoMap.Replace(old, new)
+	}
+
+	_ = validatorsInfoMap.ReplaceValidatorByKey(old.GetPublicKey(), new, old.GetShardId())
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply epochStart/metachain/legacySystemSCs.go:771]: added a log

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by mariusmihaic — APPROVED** [review]

**Pull request merged**